src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Test
+    void register_ageEquals18_ok() {
+        user.setAge(18);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginExist_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin("validLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginExist_Ok() {
+        User existingUser = new User();
+        existingUser.setLogin("differentLogin");
+        Storage.people.add(existingUser);
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginLength_notOk() {
+        user.setAge(20);
+        user.setLogin("login");
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginLength_Ok() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_PasswordLength_notOk() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("pass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_PasswordLength_Ok() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_LoginNotNull_notOk() {
+        user.setAge(20);
+        user.setLogin(null);
+        user.setPassword("validPass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
+    }
+
+    @Test
+    void register_LoginNotNull_Ok() {
+        user.setAge(20);
+        user.setLogin("validLogin");
+        user.setPassword("validPass");
+        User registeredUser = registrationServiceImpl.register(user);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void register_PasswordNotNull_notOk() {
+        user.setAge(20);
+        user.setLogin(null);
+        user.setPassword("pass");
+        assertThrows(InvalidUserDataException.class, () -> registrationServiceImpl.register(user));
```

Comment:
This test seems to be checking for a non-null password, but the user's login is set to null. This could lead to confusion about which parameter is actually being tested. Please correct the test to focus on the password parameter.